src/core/Board.ts

```diff
@@ -71,6 +73,31 @@ class Board {
     BoardInitializer.init(this);
   }
 
+  findKing(color: Color): King {
+    const pieces = this.pieces;
+
+    for(const piece of pieces) {
+        if(piece.color == color && piece.type == PieceType.King) {
+            return piece;
+        }
+    }
```

Comment:
This can be done using Array.find method:
```
const king = this.pieces.find(piece => piece.type === PieceType.King && piece.color === color)
if (!king) {
  throw new Error...
}

return king;
```